### ISSUE TITLE
Bug: Warning "expected valid finite number and received "NaN" (number)"

### ISSUE THREAD

**Issue by ukiews** [OpenWonderLabs/homebridge-switchbot#1007]
### Describe the Bug

I'm happy to report that BLE curtain 3 control is now working with the latest version of plugin and homebridge on my Pi4.
However, there is a warning that keeps getting generated:

```
[7/15/2024, 12:02:11 PM] [SwitchBot] Curtain3: Right-Curtain 3 TargetPostion: 89 sent over SwitchBot BLE,  sent successfully
[7/15/2024, 12:02:14 PM] [SwitchBot] Curtain3: Right-Curtain 3 TargetPostion: 89 sent over SwitchBot BLE,  sent successfully
[7/15/2024, 12:02:14 PM] [SwitchBot] Curtain3: Right-Curtain 3 Checking Status ...
[7/15/2024, 12:02:19 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Current Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/15/2024, 12:02:19 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Target Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/15/2024, 12:02:24 PM] [SwitchBot] Curtain3: Right-Curtain 3 ad: {
  "id": "eb8c4803381a",
  "address": "eb:8c:48:03:38:1a",
  "rssi": -79,
  "serviceData": {
    "model": "{",
    "modelName": "WoCurtain3",
    "modelFriendlyName": "Curtain 3",
    "calibration": true,
    "battery": 45,
    "inMotion": true,
    "position": 39,
    "lightLevel": 1,
    "deviceChain": 1
  }
}
```
This seems to be related to the curtain controller no responding immediately to the plugin status request. It seems there needs to be a bit more delay added before the status is requested... Or "TargetPosition" is not being properly reported.

### Device and Model

Raspberry Pi 4

### Node.js Version

20.11.0

### NPM Version

10.2.4

### Homebridge Version

1.8.3

### Homebridge Switchbot Plugin Version

3.6.0

### Homebridge Config UI X Plugin Version

4.56.4

### Operating System

Raspbian

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.6.1-beta.15/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by ukiews** [comment]
Still getting the same error. I also needed to update the 1.8.4 beta version of HB per plugin warning.

```
[7/16/2024, 10:36:23 AM] [SwitchBot] Curtain3: RightCurtain TargetPostion: 40 sent over SwitchBot BLE,  sent successfully
[7/16/2024, 10:36:26 AM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Current Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
```

Another issue I'm seeing is sending a command to two devices at the same time.
I have a left and right curtain on the same window in "Room 1." If I instruct siri to "close Room 1" only one curtain will close. Only after saying it the second time will the other curtain close. The same goes for automations to open and close "Room 1" curtains for sunrise and sunset. Only one will initially respond. Sometimes left, sometimes right. However, individual control works just fine. I'm wondering if this is an issue with the BLE signal and if there needs to be a slight delay when the signal is sent out to control multiple devices at the same time. Otherwise, it might flood the BLE receiver on the curtain 3 (?)

Example:
```
[7/16/2024, 10:38:38 AM] [SwitchBot] Curtain3: RightCurtain TargetPostion: 100 sent over SwitchBot BLE,  sent successfully
[7/16/2024, 10:38:40 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:40 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:41 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:41 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:42 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:42 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:43 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:43 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:44 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:44 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:45 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:45 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:46 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:46 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:47 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:47 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:48 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:48 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:49 AM] [SwitchBot] Curtain3: LeftCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/16/2024, 10:38:49 AM] [SwitchBot] Curtain3: LeftCurtain Retrying
[7/16/2024, 10:38:50 AM] [SwitchBot] Curtain3: LeftCurtain failed BLEpushChanges with BLE Connection, Error Message: "Cannot read properties of undefined (reading 'runToPos')"
```

**Comment by dnicolson** [comment]
I'm also seeing these warnings still on `v3.7.0`:

```
[23/07/2024, 20:11:41] [SwitchBot] Curtain: Curtain TargetPostion: 0 sent over SwitchBot BLE,  sent successfully
[23/07/2024, 20:11:51] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Current Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[23/07/2024, 20:11:51] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Target Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
```

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.7.1-beta.3/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by ukiews** [comment]
Still having the same issues where only one curtain will open/close if the command is sent to open both at once.

In this case below, I first open each one independently, then issue a command to close both.
First scenario is successful, second only Left curtain closes.

```
[7/24/2024, 1:33:44 PM] [SwitchBot] Curtain3: LeftCurtain TargetPostion: 100 sent over SwitchBot BLE,  sent successfully
[7/24/2024, 1:34:28 PM] [SwitchBot] Curtain3: RightCurtain TargetPostion: 100 sent over SwitchBot BLE,  sent successfully
[7/24/2024, 1:34:44 PM] [SwitchBot] Curtain3: LeftCurtain TargetPostion: 0 sent over SwitchBot BLE,  sent successfully
[7/24/2024, 1:34:45 PM] [SwitchBot] Curtain3: RightCurtain Checking Status ...
[7/24/2024, 1:34:47 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:47 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:48 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:48 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:49 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:49 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:50 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:50 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:51 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:51 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:52 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:52 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:53 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:53 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:54 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:54 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:55 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:55 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:56 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:56 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:57 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:57 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:57 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Current Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/24/2024, 1:34:57 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Target Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/24/2024, 1:34:58 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:58 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:34:59 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:34:59 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:35:00 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:35:00 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:35:01 PM] [SwitchBot] Curtain3: RightCurtain TypeError: Cannot read properties of undefined (reading 'runToPos')
[7/24/2024, 1:35:01 PM] [SwitchBot] Curtain3: RightCurtain Retrying
[7/24/2024, 1:35:02 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Current Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/24/2024, 1:35:02 PM] [@switchbot/homebridge-switchbot] This plugin generated a warning from the characteristic 'Target Position': characteristic value expected valid finite number and received "NaN" (number). See https://homebridge.io/w/JtMGR for more info.
[7/24/2024, 1:35:02 PM] [SwitchBot] Curtain3: RightCurtain failed BLEpushChanges with BLE Connection, Error Message: "Cannot read properties of undefined (reading 'runToPos')"
```

**Comment by dnicolson** [comment]
The original issue appears to be fixed in `v3.7.1-beta.3`.

**Comment by ukiews** [comment]
> The original issue appears to be fixed in `v3.7.1-beta.3`.

The same "Nan" and 'runToPos' errors are still appearing.

**Comment by ukiews** [comment]
Same issues with latest beta7

**Comment by ukiews** [comment]
@donavanbecker any update on this? It's so close to working smoothly!

**Comment by donavanbecker** [comment]
Sorry not yet.

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.7.1-beta.23/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by ukiews** [comment]
Still same issue. Running 3.8
When trying to control both at the same time, only one will respond, even if I set retry interval to something high like 25. But then when trying to control them one at a time, they both respond just fine.

**Comment by donavanbecker** [comment]
Can you try the latest beta: 

[![npm version](https://badgen.net/badge/homebridge-switchbot@beta/>=3.8.2-beta.1/orange)](https://www.npmjs.com/package/@switchbot/homebridge-switchbot)

[Click here to find out how to install Beta](https://github.com/OpenWonderLabs/homebridge-switchbot/wiki/Beta-Version)

**Comment by ukiews** [comment]
With 3.8.1 and 3.8.2 beta above the plugin just keeps restarting
Getting this error:

```
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:106:45: error: ‘is_lvalue_reference_v’ is not a member of ‘std’; did you mean ‘is_lvalue_reference’?
  106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
      |                                             ^~~~~~~~~~~~~~~~~~~~~
      |                                             is_lvalue_reference
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:106:66: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
  106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
      |                                                                  ^
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:106:67: error: template argument 1 is invalid
  106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
      |                                                                   ^
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:106:71: error: expected unqualified-id before ‘>’ token
  106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
      |                                                                       ^
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:123:43: error: ‘is_lvalue_reference_v’ is not a member of ‘std’; did you mean ‘is_lvalue_reference’?
  123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
      |                                           ^~~~~~~~~~~~~~~~~~~~~
      |                                           is_lvalue_reference
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:123:64: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
  123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
      |                                                                ^
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:123:65: error: template argument 1 is invalid
  123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
      |                                                                 ^
/home/pi/.cache/node-gyp/20.17.0/include/node/v8-maybe.h:123:70: error: expected unqualified-id before ‘=’ token
  123 | emplate <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
      |                                                                     ^

In file included from /home/pi/.cache/node-gyp/20.17.0/include/node/node_buffer.h:25,
                 from ../src/node_usb.h:14,
                 from ../src/node_usb.cc:1:
/home/pi/.cache/node-gyp/20.17.0/include/node/node.h:540:32: error: ‘std::string_view’ has not been declared
  540 |   static Pointer FromBlob(std::string_view in);
      |                                ^~~~~~~~~~~
/home/pi/.cache/node-gyp/20.17.0/include/node/node.h:696:8: error: ‘optional’ in namespace ‘std’ does not name a template type
  696 |   std::optional<std::string> builder_script_path;
      |        ^~~~~~~~
/home/pi/.cache/node-gyp/20.17.0/include/node/node.h:696:3: note: ‘std::optional’ is only available from C++17 onwards
  696 |   std::optional<std::string> builder_script_path;
      |   ^~~
/home/pi/.cache/node-gyp/20.17.0/include/node/node.h:768:10: error: ‘std::string_view’ has not been declared
  768 |     std::string_view main_script_source_utf8,
      |          ^~~~~~~~~~~
../src/node_usb.cc: In function ‘Napi::Value SetDebugLevel(const Napi::CallbackInfo&)’:
../src/node_usb.cc:118:71: warning: ‘void libusb_set_debug(libusb_context*, int)’ is deprecated: Use libusb_set_option instead [-Wdeprecated-declarations]
  118 |  libusb_set_debug(usb_context, info[0].As<Napi::Number>().Int32Value());
      |                                                                       ^
In file included from ../src/node_usb.h:11,
                 from ../src/node_usb.cc:1:
../libusb/libusb/libusb.h:1325:18: note: declared here
 1325 | void LIBUSB_CALL libusb_set_debug(libusb_context *ctx, int level);
      |                  ^~~~~~~~~~~~~~~~
make: *** [usb_bindings.target.mk:117: Release/obj.target/usb_bindings/src/node_usb.o] Error 1
make: Leaving directory '/var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/node_modules/usb/build'
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack at ChildProcess.<anonymous> (/opt/homebridge/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:209:23)
gyp ERR! System Linux 6.1.21-v8+
gyp ERR! command "/opt/homebridge/bin/node" "/opt/homebridge/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/node_modules/usb
gyp ERR! node -v v20.17.0
gyp ERR! node-gyp -v v10.1.0
gyp ERR! not ok 
npm error code 1
npm error path /var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/node_modules/usb
npm error command failed
npm error command sh -c node-gyp-build
npm error A complete log of this run can be found in: /home/pi/.npm/_logs/2024-09-14T23_59_24_886Z-debug-0.log
Failed to rebuild.
Restoring previous version...
Restore Complete. Installation failed.
```

**Comment by ukiews** [comment]
So I've updated my pi4 from Debian 11 to 12 (bookworm). This appears to have resolved the install errors.
However, the plugin still keeps restarting with 3.8.1

```
[9/19/2024, 12:41:56 PM] RightCurtain B6D0 is running on port 44425.
[9/19/2024, 12:41:56 PM] Please add [RightCurtain B6D0] manually in Home app. Setup Code: 282-46-114
[9/19/2024, 12:41:56 PM] LeftCurtain 4F1B is running on port 43149.
[9/19/2024, 12:41:56 PM] Please add [LeftCurtain 4F1B] manually in Home app. Setup Code: 282-46-114
TypeError: Cannot read properties of undefined (reading 'moving')
    at Curtain.getCurrentPostion (file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:826:48)
    at Curtain.BLEparseStatus (file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:381:5)
    at file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:491:11
[9/19/2024, 12:42:02 PM] [SwitchBot] Child bridge process ended
[9/19/2024, 12:42:02 PM] [SwitchBot] Process Ended. Code: 1, Signal: null
[9/19/2024, 12:42:09 PM] [SwitchBot] Restarting Process...
[9/19/2024, 12:42:09 PM] [SwitchBot] Launched child bridge with PID 3801
[9/19/2024, 12:42:12 PM] Registering platform '@switchbot/homebridge-switchbot.SwitchBot'
[9/19/2024, 12:42:12 PM] [SwitchBot] Loaded @switchbot/homebridge-switchbot v3.8.1 child bridge successfully
[9/19/2024, 12:42:12 PM] Loaded 0 cached accessories from cachedAccessories.0EE5E83C8FF8.
[9/19/2024, 12:42:12 PM] [SwitchBot] Secondary Curtain: RightCurtain, deviceId: eb8c4803381a
[9/19/2024, 12:42:12 PM] [SwitchBot] Loading external accessory: RightCurtain deviceId: eb8c4803381a
[9/19/2024, 12:42:12 PM] [SwitchBot] Secondary Curtain: LeftCurtain, deviceId: db7d1d8d184b
[9/19/2024, 12:42:12 PM] [SwitchBot] Loading external accessory: LeftCurtain deviceId: db7d1d8d184b
[9/19/2024, 12:42:13 PM] Homebridge v1.8.4 (HAP v0.12.2) (SwitchBot) is running on port 34197.
[9/19/2024, 12:42:13 PM] 

NOTICE TO USERS AND PLUGIN DEVELOPERS
> Homebridge 2.0 is on the way and brings some breaking changes to existing plugins.
> Please visit the following link to learn more about the changes and how to prepare:
> https://github.com/homebridge/homebridge/wiki/Updating-To-Homebridge-v2.0

[9/19/2024, 12:42:13 PM] RightCurtain B6D0 is running on port 42733.
[9/19/2024, 12:42:13 PM] Please add [RightCurtain B6D0] manually in Home app. Setup Code: 282-46-114
[9/19/2024, 12:42:13 PM] LeftCurtain 4F1B is running on port 33583.
[9/19/2024, 12:42:13 PM] Please add [LeftCurtain 4F1B] manually in Home app. Setup Code: 282-46-114
TypeError: Cannot read properties of undefined (reading 'moving')
    at Curtain.getCurrentPostion (file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:826:48)
    at runNextTicks (node:internal/process/task_queues:60:5)
    at listOnTimeout (node:internal/timers:545:9)
    at processTimers (node:internal/timers:519:7)
    at Curtain.BLEparseStatus (file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:381:5)
    at file:///var/lib/homebridge/node_modules/@switchbot/homebridge-switchbot/src/device/curtain.ts:491:11
```

The same thing is happening on newer versions.
3.8.0 is stable in this regard.

Of course, curtains still don't close together. Only one responds. It also appears that the one that does respond is the one that was last controlled.

**Comment by donavanbecker** [comment]
Closing all issues now that v4.0.0 is out. If you are still having an issue after trying v4.0.0, please open a new issue. Keep in mind v4.0.0 has some breaking changes with the config. You will need to re set your config up, however there are some nice added bonuses since you can set settings at a deviceType level or individual level.
